src/playback/view/playbacktoolbarmodel.cpp

```diff
@@ -74,13 +82,15 @@ void PlaybackToolBarModel::setupConnections()
         onActionsStateChanges({ PLAY_ACTION_CODE });
     });
 
-    playbackController()->playbackPositionChanged().onNotify(this, [this]() {
-        updatePlayPosition();
+    player->playbackPositionChanged().onReceive(this, [this](secs_t secs) {
+        updatePlayPosition(secs);
     });
 
-    playbackController()->totalPlayTimeChanged().onNotify(this, [this]() {
+    playbackController()->totalPlayTimeChanged().onNotify(this, [this, player]() {
```

Comment:
It would be better to call globalContext()->currentPlayer() again inside the lambda instead of capturing the player. Or you can use week_pointer here. This way it is easier to control the pointer lifetime (it won't live longer than it should)

Reply:
fixed